File: Samples/visio_managed_code_interop_vs2010/VS1010_CSharp_Samples/shape_getresults.cs
using IVisio = Microsoft.Office.Interop.Visio;

public static partial class VS2010_CSharp_Samples
{
    public static void Shape_GetResults(IVisio.Document doc)
    {
        var page = VisioInterop.Util.CreateStandardPage(doc, "SGR");
        var shape = VisioInterop.Util.CreateStandardShape(page);

        // CREATE REQUEST
        var request = new[]
        {
              new
                  {
                      Section = (short)IVisio.VisSectionIndices.visSectionObject, 
                      Row=(short)IVisio.VisRowIndices.visRowXFormOut, 
                      Cell=(short)IVisio.VisCellIndices.visXFormWidth,
                      UnitCode=(short) IVisio.VisUnitCodes.visNoCast
                  },                        
              new
                  {
                      Section = (short)IVisio.VisSectionIndices.visSectionObject, 
                      Row=(short)IVisio.VisRowIndices.visRowXFormOut, 
                      Cell=(short)IVisio.VisCellIndices.visXFormHeight,
                      UnitCode=(short) IVisio.VisUnitCodes.visNoCast
                  }                        
        };

        // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        var SRCStream = new short[request.Length * 3];
        var unitcodes = new object[request.Length];
        for (int i = 0; i < request.Length; i++)
        {
            SRCStream[(i * 3) + 0] = request[i].Section;
            SRCStream[(i * 3) + 1] = request[i].Row;
            SRCStream[(i * 3) + 2] = request[i].Cell;
            unitcodes[i] = request[i].UnitCode;
        }

        // EXECUTE THE REQUEST
        short flags = (short)IVisio.VisGetSetArgs.visGetFloats;
        System.Array results_sa = null;
        shape.GetResults(SRCStream, flags, unitcodes, out results_sa);

        // MAP OUTPUT BACK TO SOMETHING USEFUL 
        var results_doubles = new double[results_sa.Length];
        results_sa.CopyTo(results_doubles, 0);

        // DISPLAY THE INFORMATION
        shape.Text = string.Format("Results={0},{1}", results_doubles[0], results_doubles[1]);
    }
}

File: Samples/visio_managed_code_interop_vs2010/Visio2010_VB_Samples/Shape_GetResults.vb
Imports IVisio = Microsoft.Office.Interop.Visio


Partial Public Class VS2010_VB_Samples
    Shared Sub Shape_GetResults (ByVal doc As Microsoft.Office.Interop.Visio.Document)
        Dim page = VisioInterop.Util.CreateStandardPage (doc, "SGR")
        Dim shape = VisioInterop.Util.CreateStandardShape (page)

        ' CREATE REQUEST
        Dim request = {New With { _
                Key .Section = CShort (IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort (IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort (IVisio.VisCellIndices.visXFormWidth), _
                Key .UnitCode = CShort (IVisio.VisUnitCodes.visNoCast) _
                }, New With { _
                Key .Section = CShort (IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort (IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort (IVisio.VisCellIndices.visXFormHeight), _
                Key .UnitCode = CShort (IVisio.VisUnitCodes.visNoCast) _
                }}

        ' MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        Dim SRCStream = New Short(request.Length*3 - 1) {}
        Dim unitcodes = New Object(request.Length - 1) {}
        For i As Integer = 0 To request.Length - 1
            SRCStream ((i*3) + 0) = request (i).Section
            SRCStream ((i*3) + 1) = request (i).Row
            SRCStream ((i*3) + 2) = request (i).Cell
            unitcodes (i) = request (i).UnitCode
        Next

        ' EXECUTE THE REQUEST
        Dim flags As Short = CShort (IVisio.VisGetSetArgs.visGetFloats)
        Dim results_sa As System.Array = Nothing
        shape.GetResults (SRCStream, flags, unitcodes, results_sa)

        ' MAP OUTPUT BACK TO SOMETHING USEFUL 
        Dim results_doubles = New Double(results_sa.Length - 1) {}
        results_sa.CopyTo (results_doubles, 0)

        ' DISPLAY THE INFORMATION
        shape.Text = String.Format ("Results={0},{1}", results_doubles (0), results_doubles (1))

    End Sub
End Class
